components/collector/src/collector_utilities/date_time.py

```diff
@@ -28,6 +31,17 @@ def parse_datetime(text: str) -> datetime:
     return date_time.replace(tzinfo=tzlocal()) if date_time.tzinfo is None else date_time
 
 
+def parse_duration(text: str) -> int:
+    """Parse the duration in minutes from the text. Expects '<nr>h <nr>m <nr>s' with all groups optional."""
+    regexp = r"(?:(?P<hours>\d+)h\s*)?(?:(?P<minutes>\d+)m\s*)?(?:(?P<seconds>\d+)s\s*)?"
```

Comment:
This feels like something you might waste a while on when trying to debug. The `\s*` can be brought out of the groups and I don't think the non-capturing groups are needed, since only named groups are looked up. Something like this should provide the same functionality:
```
rs = r"((?P<seconds>\d+)s)?"
rm = r"((?P<minutes>\d+)m)?"
rh = r"((?P<hours>\d+)h)?"
regexp = r"\s*".join((rh, rm, rs))
```